spark/src/main/java/org/apache/iceberg/spark/source/BatchDataReader.java

```diff
@@ -75,6 +103,16 @@
       }
 
       iter = builder.build();
+    } else if (task.file().format() == FileFormat.ORC) {
+      iter = ORC.read(location)
+          .project(expectedSchema)
```

Comment:
I will change this to not project constant columns to avoid materializing a `ColumnVector` similar to what we did in #1191

Reply:
Fixed in https://github.com/apache/iceberg/pull/1189/commits/9c2af8d5038b8dd478c39f57729370f7244816a7